tico/interpreter/infer.py

```diff
@@ -45,15 +47,43 @@ def preprocess_inputs(inputs: Any):
     return tuple(l)
 
 
+def flatten_and_convert(inputs: Any) -> tuple:
```

Comment:
```suggestion
def flatten_and_convert(inputs: Sequence) -> tuple:
```

And `preprocess_inputs` definition can be removed.